src/app-layer-htp-file.c

```diff
@@ -145,6 +145,20 @@ int HTPFileOpen(HtpState *s, const uint8_t *filename, uint16_t filename_len,
 
     FileSetTx(files->tail, txid);
 
+    htp_tx_t *tx = AppLayerParserGetTx(IPPROTO_TCP, ALPROTO_HTTP, s, txid);
+    if (tx) {
+        switch (tx->request_method_number) {
+            case HTP_M_POST:
```

Comment:
Just realized we can have a POST request that get a file in result so this is not correct.